### PR TITLE
Remove Micropayment Channels.

### PR DESCRIPTION
**Pull request by schildbach**
Now that HTLC-based bi-directional payment channels are state of the art, nobody will ever use the old style. Let's remove them in order to reduce maintenance for bitcoinj.

### CODE REVIEW

File: core/src/test/java/org/bitcoinj/net/discovery/SeedPeersTest.java [hunk 1]
```diff
@@ -1,55 +0,0 @@
...
-        assertThat(seedPeers.getPeer(), equalTo(null));
-    }
-    
-    @Test
-    public void getPeers_length() throws Exception{
-        SeedPeers seedPeers = new SeedPeers(MAINNET);
-        InetSocketAddress[] addresses = seedPeers.getPeers(0, 0, TimeUnit.SECONDS);
-        assertThat(addresses.length, equalTo(MAINNET.getAddrSeeds().length));
-    }
-}
```
Comment (msgilligan) [inline core/src/test/java/org/bitcoinj/net/discovery/SeedPeersTest.java]: Are these tests related to Payment Channels? Did you mean to remove this one?

File: core/src/test/java/org/bitcoinj/net/discovery/DnsDiscoveryTest.java [hunk 2]
```diff
@@ -1,60 +0,0 @@
...
-        assertNotEquals(0, inetSocketAddresses.length);
-    }
-
-    @Test(expected = PeerDiscoveryException.class)
-    public void testGetPeersThrowsPeerDiscoveryExceptionForUnknownHost() throws PeerDiscoveryException {
-        DnsDiscovery.DnsSeedDiscovery dnsSeedDiscovery = new DnsDiscovery.DnsSeedDiscovery(MainNetParams.get(),
-                "unknown host");
-        dnsSeedDiscovery.getPeers(0, 100, TimeUnit.MILLISECONDS);
-    }
-}
```
Comment (msgilligan) [inline core/src/test/java/org/bitcoinj/net/discovery/DnsDiscoveryTest.java]: Did you mean to remove this one?

**Comment by msgilligan** [comment]
It looks like you accidentally removed two tests that aren't related to payment channels. Other that that, this is a lot of unused code, so it makes sense to remove it.

On a separate note, I'm interested in implementing Lightning payment channels -- maybe we should discuss that elsewhere?

File: core/src/test/java/org/bitcoinj/net/discovery/SeedPeersTest.java (hunk 1 above)
Comment (schildbach) [reply core/src/test/java/org/bitcoinj/net/discovery/SeedPeersTest.java]: You're right, thanks! I added the two classes back.

**Comment by schildbach** [comment]
Yes, that would be great to have. We should discuss it on the mailing list.

**Comment by schildbach** [comment]
Merged.
